docs/introduction/release_process.md

```diff
@@ -23,6 +23,7 @@ They can include:
 - Deprecation of functionality (with an alternative functionality provided).
 - Configuration changes.
 
+Occasionally, new features aren't added directly through a feature release. Instead, a tag is created on the feature branch to informally release the feature. Tags are named following the same convention as formal releases. 
```

Comment:
I'm not sure this is correct. Tags on the feature branches will not necessarily be of the form mbed-os-x.y.z as they are not official releases. They should use semantic versioning but I'm not sure we have defined a specific convention for these ....

Reply:
using `feature-` prefix (definitely not following `mbed-os-` as its not official release).